### PR TITLE
enhance(build): return build on CreateBuild and UpdateBuild

### PR DESCRIPTION
**Pull request by ecrupper**
Part of the effort for https://github.com/go-vela/community/issues/613. It would be helpful to not have to make two calls to the database during our webhook process. This can be remedied by returning the created build object when it's saved to the database.

### CODE REVIEW

**Review by cognifloyd — APPROVED** [review]
Looks sane

**Review by cognifloyd — APPROVED** [review]

File: database/postgres/build.go [hunk 1]
```diff
@@ -156,17 +158,25 @@ func (c *client) CreateBuild(b *library.Build) error {
...
-	return c.Postgres.
+	err = c.Postgres.
 		Table(constants.TableBuild).
 		Create(build.Crop()).Error
+
+	if err != nil {
+		return nil, err
+	}
+
+	return build.Crop().ToLibrary(), nil
```
Comment (jbrockopp) [inline database/postgres/build.go:173]: Have we confirmed the `build` object has the ID field populated with this change?

We don't explicitly set that ID field in our codebase which is why I ask.

**Review by jbrockopp — CHANGES_REQUESTED** [review]

File: database/postgres/build.go (hunk 1 above)
Comment (cognifloyd) [reply database/postgres/build.go:173]: When gorm saves the object, it populates the ID field, right?

**Comment by ecrupper** [comment]
closing because this has become a mess. Will just use a new branch

**Pull request closed without merging**